From aaf55113bb7174787ffe8aa6878086943238a63e Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Thu, 3 Jan 2013 23:19:19 +0100
Subject: [PATCH 084/482] 	* util/grub.d/10_kfreebsd.in: Correct the
 patch to zpool.cache as it's 	always in /boot/zfs. 	Reported by: Yuta
 Satoh.

---
 ChangeLog                  | 6 ++++++
 util/grub.d/10_kfreebsd.in | 4 ++--
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 936af2f..a28a1f7 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,9 @@
+2013-01-03  Vladimir Serbinenko  <phcoder@gmail.com>
+
+	* util/grub.d/10_kfreebsd.in: Correct the patch to zpool.cache as it's
+	always in /boot/zfs.
+	Reported by: Yuta Satoh.
+
 2013-01-03  Yuta Satoh  <nigoro>
 
 	* util/grub.d/10_kfreebsd.in: Fix improper references to grub-probe by
diff --git a/util/grub.d/10_kfreebsd.in b/util/grub.d/10_kfreebsd.in
index 2a48b52..c123ceb 100644
--- a/util/grub.d/10_kfreebsd.in
+++ b/util/grub.d/10_kfreebsd.in
@@ -122,10 +122,10 @@ EOF
     zfs)
       load_kfreebsd_module opensolaris false
 
-      ls "${dirname}/zfs/zpool.cache" > /dev/null
+      ls "/boot/zfs/zpool.cache" > /dev/null
       printf '%s\n' "${prepare_boot_cache}"
       sed "s/^/$submenu_indentation/" << EOF
-	kfreebsd_module		${rel_dirname}/zfs/zpool.cache type=/boot/zfs/zpool.cache
+	kfreebsd_module		$(make_system_path_relative_to_its_root /boot)/zfs/zpool.cache type=/boot/zfs/zpool.cache
 EOF
     ;;
   esac
-- 
1.8.2.1

